File: AnagramMain/AnagramException.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AnagramMain
{
	public class AnagramException : Exception
	{
		public string Word { get; }

		public int LineCount { get; }

		public int WordCount { get; }

		public AnagramException(string message, string word, int lineCount = 0, int wordCount = 0) : base(message)
		{
			Word = word;
			LineCount = lineCount;
			WordCount = wordCount;
		}
	}
}

File: AnagramUtils/AnagramException.cpp
#include "pch.h"
#include <string>
#include "AnagramException.h"

using namespace std;

AnagramException::AnagramException(string message, string word, int lineCount /*= 0*/, int wordCount /*= 0*/)
	: Message(message), Word(word), LineCount(lineCount), WordCount(wordCount)
{
}

AnagramException::~AnagramException() { }

string AnagramException::GetExMessage() { return Message; }

string AnagramException::GetWord() { return Word; }

int AnagramException::GetLineCount() { return LineCount; }

int AnagramException::GetWordCount() { return WordCount; }

